src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,305 @@
+     * 
+     * @param input The user command.
+     * @return A {@link Command} instance created according to the user command.
+     * @throws DukeException Thrown when the command given
+     *                       is unrecognized.
+     */
+    public Command parse(String input) throws DukeException {
+        // Ignore empty user input
+        if (input.equals("")) {
+            return new EmptyCommand();
+        }
+
+        // Extract main command first
+        String command = input.split(" ")[0];
+
+        // Parse main command
+        switch (command) {
+        case "bye":
+            return new ExitCommand();
+        case "list":
+            return new ListCommand();
+        case "mark":
+            return parseMarkCommand(input, "mark");
+        case "unmark":
+            return parseMarkCommand(input, "unmark");
+        case "todo":
+            return parseTodoCommand(input);
+        case "deadline":
+            return parseDeadlineCommand(input);
+        case "event":
+            return parseEventCommand(input);
+        case "delete":
+            return parseDeleteCommand(input);
+        case "find":
+            return parseFindCommand(input);
+        default:
+            throw new DukeException(new String[] {
+                "Unrecognized command " + Ui.cTxt(command, Ui.Color.PURPLE),
+                "Maybe create a new TODO with "
+                        + Ui.cTxt("todo", Ui.Color.PURPLE)
+                        + " read a book?"
+            });
+        }
+    }
+
+    /**
+     * A method to extract and join the elements from index 1 to the end.
+     * Used for situations where it is needed to extract 
+     * "read a book" from "todo read a book".
+     * 
+     * @param item The array containing the contents to be extracted.
+     * @return A string that contains the extracted contents 
+     *         joined by empty space.
+     */
+    private static String extractTail(String[] item) {
+        return String.join(" ",
+            Arrays.copyOfRange(
+                item, 1, item.length
+            )
+        );
+    }
+
+    /**
+     * Parses the content of a "mark" or "unmark" command.
+     * Checks if a valid number is provided.
+     * 
+     * @param input The mark command.
+     * @param type Indicates whether it is parsing 
+     *             a mark or unmark command.
+     * @return A {@link MarkCommand} instance.
+     * @throws DukeException Thrown when no number is given.
+     */
+    private Command parseMarkCommand(String input, String type) throws DukeException {
+        String[] parseArr = input.split(" ");
+        if (parseArr.length < 2) {
+            throw new DukeException(new String[] {
+                "Looks like you're missing a number:",
+                "Try " + Ui.cTxt("mark", Ui.Color.PURPLE) + " 1"
+            });
+        }
+
+        return new MarkCommand(parseArr[1], type == "mark");
```

Comment:
In the parseMarkCommand() method, comparing strings using == (like type == "mark") is not ideal. Instead, use type.equals("mark") for string comparisons.